src/main/java/org/elasticsearch/action/admin/indices/syncedflush/SyncedFlushIndicesRequest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.syncedflush;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * A synced flush request to sync flush one or more indices.
+ * <p>Best created with {@link org.elasticsearch.client.Requests#flushRequest(String...)}.
+ */
+public class SyncedFlushIndicesRequest extends ActionRequest implements IndicesRequest.Replaceable {
+
+    private String[] indices;
+
+    // timeout in ms
+    long timeout = 30000;
```

Comment:
can we have a timeout that is consistent with other ie. being a TimeValue instead? I also wonder if this one shoudl have a timeout at all? I think it shouldn't but rather be like _flush?